### PR TITLE
[SPARK-48504][PYTHON][CONNECT] Parent Window class for Spark Connect and Spark Classic

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
 Parent Window class for Spark Connect and Spark Classic

### Why are the changes needed?
Same as https://github.com/apache/spark/pull/46129

### Does this PR introduce _any_ user-facing change?
Same as https://github.com/apache/spark/pull/46129

### How was this patch tested?
CI

### Was this patch authored or co-authored using generative AI tooling?
NO

### CODE REVIEW

File: python/pyspark/sql/connect/window.py [hunk 1]
```diff
@@ -18,24 +18,23 @@
...
-from pyspark.util import (
-    JVM_LONG_MIN,
-    JVM_LONG_MAX,
-)
 from pyspark.sql.window import Window as PySparkWindow, WindowSpec as PySparkWindowSpec
 from pyspark.errors import PySparkTypeError
 
 if TYPE_CHECKING:
-    from pyspark.sql.connect._typing import ColumnOrName
+    from pyspark.sql.connect._typing import ColumnOrName, ColumnOrName_
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/window.py:37]: Why do we need `ColumnOrName_`?

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/sql/connect/window.py (hunk 1 above)
Comment (zhengruifeng) [reply python/pyspark/sql/connect/window.py:37]: This change is to make the typing consistent with classic side, which uses `ColumnOrName_`.

The `ColumnOrName_` was introduced in https://github.com/apache/spark/commit/a70006d9a7b578721d152d0f89d1a894de38c25d
 and used in multiple places. Probably we can revisit it and remove it if possible.

Comment (HyukjinKwon) [reply python/pyspark/sql/connect/window.py:37]: ah okie. That's fine.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**